### PR TITLE
[feat]: [CDS-79477]: Repo-filter support for List API 

### PR DESCRIPTION
**Pull request by senjucanon2**
## Describe your changes:
Changes: Have added repoName foe filter support in list Env API.
Testing : Env were filtered in response according to the repoName provided. Also repoName filter was used with other filter params, and expected results were observed.

## Checklist
- [ ] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/53804)

### CODE REVIEW

**Comment by senjucanon2** [comment]
trigger uts

**Comment by krishi0408** [comment]
trigger gitleaks

**Comment by senjucanon2** [comment]
trigger uts

**Comment by senjucanon2** [comment]
trigger uts

File: 127-cd-nextgen-entities/src/test/java/io/harness/ng/core/environment/mappers/EnvironmentFilterHelperTest.java [hunk 1]
```diff
@@ -136,8 +137,8 @@ public void testGetUpdateOperationsForDelete() {
   public void testListWithNamesFilter() {
     EnvironmentFilterPropertiesDTO environmentFilterPropertiesDTO =
         EnvironmentFilterPropertiesDTO.builder().environmentNames(Arrays.asList("qa", "dev")).build();
-    Criteria criteria = environmentFilterHelper.createCriteriaForGetList(
-        accountIdentifier, orgIdentifier, projectIdentifier, false, null, null, environmentFilterPropertiesDTO, false);
+    Criteria criteria = environmentFilterHelper.createCriteriaForGetList(accountIdentifier, orgIdentifier,
+        projectIdentifier, false, null, null, environmentFilterPropertiesDTO, false, Strings.EMPTY);
```
Comment (hingerabhinav) [inline 127-cd-nextgen-entities/src/test/java/io/harness/ng/core/environment/mappers/EnvironmentFilterHelperTest.java:141]: Let's use StringUtils.EMPTY by apache commons for uniformity

**Review by senjucanon2 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/53804)* status: 4 file(s) to review. 1. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`127-cd-nextgen-entities/src/test/java/io/harness/ng/core/environment/mappers/EnvironmentFilterHelperTest.java` line 141 at r3](https://reviewable.io/reviews/harness/harness-core/53804#gh-1350120877:-NgIzoK73EHepFIYGPeH:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/46e7513cee4e62ae080e81ea46387b16a0ce503d/127-cd-nextgen-entities/src/test/java/io/harness/ng/core/environment/mappers/EnvironmentFilterHelperTest.java#L141)):*
<details><summary><i>Previously, hingerabhinav (Abhinav Hinger) wrote…</i></summary><blockquote>

Let's use StringUtils.EMPTY by apache commons for uniformity 
</blockquote></details>

Done.

**Review by hingerabhinav — APPROVED** [review]

**Review by yogesh-chauhan — APPROVED** [review]
:lgtm:

> Reviewed 1 of 2 files at r1, 1 of 1 files at r2, 2 of 2 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/53804#-:-NgJ2uFU9ltA9gtAHpEQ:bnfp4nl)* status: 4 file(s) to review. 1. unresolved discussions.

**Comment by senjucanon2** [comment]
trigger uts, review

**Review by adivishy1 — COMMENTED** [review]
> Reviewed 2 of 2 files at r1, 1 of 1 files at r2, 1 of 1 files at r3.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/53804)* status: 2 file(s) to review. 1. unresolved discussions.

**Pull request merged**